src/common.rs

```diff
@@ -109,91 +109,55 @@ pub(crate) fn config_file_get() -> String {
-    }
+    config_get_env("general", "receive_revocation_port", "REVOCATION_PORT")
 }
 
 /// Returns cloud agent IP from keylime.conf if env var not present
 pub(crate) fn cloudagent_ip_get() -> Result<String> {
-    match env::var("CLOUDAGENT_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("cloud_agent", "cloudagent_ip")
-            }
-        }
-        _ => config_get("cloud_agent", "cloudagent_ip"),
-    }
+    config_get_env("cloud_agent", "cloudagent_ip", "CLOUDAGENT_IP")
 }
 
 /// Returns cloud agent port from keylime.conf if env var not present
 pub(crate) fn cloudagent_port_get() -> Result<String> {
-    match env::var("CLOUDAGENT_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("cloud_agent", "cloudagent_port")
-            }
-        }
-        _ => config_get("cloud_agent", "cloudagent_port"),
-    }
+    config_get_env("cloud_agent", "cloudagent_port", "CLOUDAGENT_PORT")
 }
 
 /// Returns registrar IP from keylime.conf if env var not present
 pub(crate) fn registrar_ip_get() -> Result<String> {
-    match env::var("REGISTRAR_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("registrar", "registrar_ip")
-            }
-        }
-        _ => config_get("registrar", "registrar_ip"),
-    }
+    config_get_env("registrar", "registrar_ip", "REGISTRAR_IP")
 }
 
 /// Returns registrar port from keylime.conf if env var not present
 pub(crate) fn registrar_port_get() -> Result<String> {
-    match env::var("REGISTRAR_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("registrar", "registrar_port")
-            }
-        }
-        _ => config_get("registrar", "registrar_port"),
+    config_get_env("registrar", "registrar_port", "REGISTRAR_PORT")
+}
+
+/// Returns the contact ip for the agent if set
+pub(crate) fn cloudagent_contact_ip_get() -> Option<String> {
+    match config_get_env(
+        "cloud_agent",
+        "agent_contact_ip",
+        "KEYLIME_AGENT_CONTACT_IP",
+    ) {
+        Ok(ip) => Some(ip),
+        Err(_) => None, // Ignore errors because this option might not be set
+    }
+}
+
+/// Returns the contact ip for the agent if set
+pub(crate) fn cloudagent_contact_port_get() -> Option<u32> {
+    match config_get_env(
+        "cloud_agent",
+        "agent_contact_port",
+        "KEYLIME_AGENT_CONTACT_PORT",
+    ) {
+        Ok(port) => Some(port.parse::<u32>().unwrap()),
```

Comment:
Afaik `.unwrap()` is not allowed in this project unless it's explicitly marked with `//#[allow_ci]`, but in this case `str::parse<u32>` actually could fail if `port` is not parsable. Maybe good to change the return type of this function to `Result<Option<u32>>`?

Reply:
I agree with @ueno .

Reply:
Fixed